daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/GrpcLedgerClient.scala

```diff
@@ -382,6 +382,17 @@ class GrpcLedgerClient(val grpcClient: LedgerClient, val applicationId: Applicat
       grpcClient.userManagementClient
         .listUsers(pageToken = pageToken, pageSize = pageSize)
         .flatMap { case (users, nextPageToken) =>
+          // A note on loop termination:
+
+          // Currently the API always returns a final page containing 0 users. When this final page
+          // is delivered, it is accompanied by an empty string as the nextPageToken.  To avoid this
+          // extra roundtrip in the common case, we test for 'users.size < pageSize'.
+
+          // The 'nextPageToken == ""' test is currently redundant because an empty nextPageToken
+          // *only* comes with the 0-users page, but we add the test anyway as a safeguard against a
+          // possible future behavior change, where a final page containing exactly 'pageSize' users
+          // is delivered along with an empty nextPageToken, and this indicates no more pages.
```

Comment:
Given what I read in the documentation, this sounds like a buggy upstream behavior.

Reply:
fixed following discussion on slack

Reply:
Looks like it's going to be solved soon (https://github.com/digital-asset/daml/pull/12773). 🙂 Thanks for uncovering this!